src/WebJobs.Script.WebHost/Security/IScriptSecretSerializer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public interface IScriptSecretSerializer
+    {
+        int SupportedFormatVersion { get; }
+
+        IList<Key> DeserializeFunctionSecrets(JObject secrets);
+
+        string SerializeFunctionSecrets(IList<Key> secrets);
```

Comment:
I'm wondering if we shouldn't have a Type for Function secrets, like we have for HostSecrets. That would make the APIs more symmetric, and you wouldn't have to update this signature if we add additional secret properties etc. for function secrets. Your file formats (with the "keys" properties, etc.) supports extensibility in the future.